### PR TITLE
Fix events emitted twice

### PR DESCRIPTION
**Pull request by clobrano**
- Handle NHC timed out event only once
- Emit RemediationFinished event only once

see:
https://issues.redhat.com/browse/ECOPROJECT-1860
https://issues.redhat.com/browse/ECOPROJECT-1861

### CODE REVIEW

File: controllers/machinedeletionremediation_controller.go [hunk 1]
```diff
@@ -133,9 +133,14 @@ func (r *MachineDeletionRemediationReconciler) Reconcile(ctx context.Context, re
 	}()
 
 	if r.isTimedOutByNHC(mdr) {
-		log.Info("NHC time out annotation found, stopping remediation")
-		commonevents.RemediationStoppedByNHC(r.Recorder, mdr)
-		_, err = r.updateConditions(remediationTimedOutByNhc, mdr)
+		updateRequired, err := r.updateConditions(remediationTimedOutByNhc, mdr)
+		if err == nil && updateRequired {
```
Comment (slintes) [inline controllers/machinedeletionremediation_controller.go:137]: for readability I would prefer this:
```
if updateRequired, err := ...; err != nil {
  return ctrl.Result{}, err
} else if updateRequired {
  ...
}
return ctrl.Result{}, nil
```
but I won't block on this... same at the other code place

**Review by slintes — COMMENTED** [review]
non blocker comment...  I leave it up to you

/lgtm
/hold

File: controllers/machinedeletionremediation_controller.go (hunk 1 above)
Comment (clobrano) [reply controllers/machinedeletionremediation_controller.go:137]: sgtm

**Pull request merged**